### ISSUE TITLE
Unexpected predictions for one token in CnnSentenceClassificationExample

### ISSUE THREAD

**Issue by avanco** [deeplearning4j/deeplearning4j#4740]
#### Issue Description

I had run CnnSentenceClassificationExample, adding just a piece of code to test the model with some sentences. Here the gist marked with BEGIN and END changes: https://gist.github.com/avanco/3ba75ad4fe0c4d0448541addbda98b1b

When trying to predict any sentence with two or more tokens it looks ok, but when input is only one word I had different results for each execution of the same input.

- expected behavior:
nice
Predictions for first negative review:
P(Negative) = some real value (x)
P(Positive) = some real value (x)

nice
Predictions for first negative review:
P(Negative) = again (x)
P(Positive) = again (x)

- encountered behavior:
nice
Predictions for first negative review:
P(Negative) = 0.5224869251251221
P(Positive) = 0.4775131046772003

nice
Predictions for first negative review:
P(Negative) = 0.48594552278518677
P(Positive) = 0.5140544772148132

#### Version Information

Please indicate relevant versions, including, if relevant:

* Deeplearning4j version: version defined in: dl4j-examples
* platform information (OS, etc): Linux 4.9.83-1-MANJARO
* CUDA version, if used --> NOT USED
* NVIDIA driver version, if in use --> NOT USED

**Comment by saudet** [comment]
You're not setting the random seed anywhere, that might have an influence...

**Comment by avanco** [comment]
I am not sure if I understand, but I tried to set a seed in NeuralNetConfiguration.Builder(), but the results are still unexpected

**Comment by AlexDBlack** [comment]
I can confirm something odd is happening here... it's reproducable on 0.9.1 without even training the network. Most of the time (for single word input) I get p=1 for one of the classes, or I get inconsistent probabilities with different iterations (same input).

**Comment by AlexDBlack** [comment]
Thanks for reporting - looks like this may have already been fixed. Not sure exactly which bug we were running into (there's been a bunch of bug fixes since 0.9.1), but it's not reproducable with master.

You can try again with snapshots. https://deeplearning4j.org/snapshots
If you do, I did notice that I needed to switch to ```WordVectorSerializer.readWord2VecModel()```  method for loading the word vectors.

Output on 0.9.1 (without training model):
```
nice
P(Negative) = 0.0
P(Positive) = 1.0
nice
P(Negative) = 0.38471877574920654
P(Positive) = 0.6152812242507935
nice
P(Negative) = 0.0
P(Positive) = 1.0
hello
P(Negative) = 0.0
P(Positive) = 1.0
hello
P(Negative) = 1.0
P(Positive) = 0.0
hello
P(Negative) = 0.0
P(Positive) = 1.0
```

Output on master (without training model):
```
nice
P(Negative) = 0.5233780145645142
P(Positive) = 0.47662195563316345
nice
P(Negative) = 0.5233780145645142
P(Positive) = 0.47662195563316345
nice
P(Negative) = 0.5233780145645142
P(Positive) = 0.47662195563316345
hello
P(Negative) = 0.538331151008606
P(Positive) = 0.46166887879371643
hello
P(Negative) = 0.538331151008606
P(Positive) = 0.46166887879371643
hello
P(Negative) = 0.538331151008606
P(Positive) = 0.46166887879371643
```
